### PR TITLE
created updatePost method in Post.vue and dispatch the updatePost act…

### LINKED ISSUE #4: Edit Post Feature

### PR DESCRIPTION
**Pull request by AsmaaAlbaik**
…ion in post.js page and commit it in updatePost mutations + installed the moment.js package to formate time and get how match time ago

Fixes #4 

## Proposed Changes

- created updatePost method in Post.vue and dispatch the
updatePost action in post.js page by use put http request
- commit it in updatePost mutations.
- in updatePost I used map method to update the postlist array
- installed the moment.js package to formate time and get how match time
 ago

  -

### CODE REVIEW

File: src/components/Timeline/Post.vue [hunk 1]
```diff
@@ -13,11 +13,47 @@
...
+          <div class="dropdown-menu" v-if="getUserId == post.author.id">
+            <button
+              class="dropdown-item"
+              type="button"
+              data-toggle="modal"
+              :data-target="'#updateModal_' + post.id"
+            >
+              Edit
+            </button>
+            <button class="dropdown-item" type="button">Delete</button>
```
Comment (muhmushtaha) [inline src/components/Timeline/Post.vue]: Where is the delete event?!

```suggestion
            <button class="dropdown-item" type="button" @click="DeletePost(post.id)">Delete</button>
```

File: package.json [hunk 2]
```diff
@@ -11,10 +11,12 @@
     "bootstrap": "^4.4.1",
     "core-js": "^3.6.4",
     "jquery": "^3.4.1",
+    "moment": "^2.24.0",
     "popper.js": "^1.16.1",
     "vue": "^2.6.11",
     "vue-resource": "^1.5.1",
     "vue-router": "^3.1.5",
+    "vue-timeago": "^5.1.2",
```
Comment (muhmushtaha) [inline package.json]: Remove the package `vue-timeago` if you aren't using it.

**Comment by muhmushtaha** [comment]
image

File: package.json (hunk 2 above)
Comment (AsmaaAlbaik) [reply package.json]: okay

File: src/components/Timeline/Post.vue (hunk 1 above)
Comment (AsmaaAlbaik) [reply src/components/Timeline/Post.vue]: it exists!!

File: src/components/Timeline/PostItem.vue [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      <p class="card-text" v-else>
+        {{ post.content }}
+        <a href="#" @click.prevent="loadSubContent()"> read less</a>
+      </p>
+    </div>
+  </div>
+</template>
+
+<script>
+var moment = require("moment");
```
Comment (muhmushtaha) [inline src/components/Timeline/PostItem.vue]: Use import instead of require.

**Comment by muhmushtaha** [comment]
Wrong scenario for edit.

editPostErr2

**Pull request merged**